epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyVisibilityItem.java

```diff
@@ -0,0 +1,156 @@
+package com.airbnb.epoxy;
+
+import android.graphics.Rect;
+import android.support.annotation.NonNull;
+import android.support.annotation.Px;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+
+class EpoxyVisibilityItem {
+
+  private final Rect localVisibleRect = new Rect();
+
+  private int adapterPosition = RecyclerView.NO_POSITION;
+
+  @Px
+  int size;
+
+  private int otherSize;
+
+  private boolean vertical;
+
+  private float percentVisibleSize = 0.f;
+
+  private int visibleSize;
+
+  private int viewportSize;
+
+  private boolean fullyVisible = false;
+  private boolean visible = false;
+  private boolean focusedVisible = false;
+
+  /** Store last value for deduping */
+  private int lastVisibleSizeNotified = -1;
+
+  /**
+   * Update the visibility item according the current layout.
+   *
+   * @param view        the current {@link com.airbnb.epoxy.EpoxyViewHolder}'s itemView
+   * @param parent      the {@link android.support.v7.widget.RecyclerView}
+   * @param orientation the {@link android.support.v7.widget.LinearLayoutManager}'s orientation
```

Comment:
Done, we can actually use `canScrollVertically()` on `LayoutManager`. 

Just understand that the visibility tracker is not really working yet when it can scroll in both direction (ex with a model in a carousel nested in a vertical list). And I would prefer address it in a different pull request. We could mention it in the documentation in the meantime.